File: fixture-app-node/format.js
/**
 * @param {number} amount
 * @returns {{status: number, body: string}}
 */
function formatOrder(amount) {
  if (typeof amount !== "number" || Number.isNaN(amount) || amount < 0) {
    return { status: 400, body: JSON.stringify({ error: "amount must be non-negative" }) };
  }
  return { status: 200, body: JSON.stringify({ amount, currency: "USD" }) };
}

module.exports = { formatOrder };

File: fixture-app-rust/src/format.rs
use crate::http_util;
use std::collections::HashMap;

/// Renders an order's amount as a JSON view, rejecting a negative amount
/// rather than echoing back a nonsensical order.
pub fn format_order(amount: i64) -> Option<String> {
    if amount < 0 {
        return None;
    }
    Some(format!("{{\"amount\":{amount},\"currency\":\"USD\"}}"))
}

pub fn handle(query: &HashMap<String, String>) -> (u16, String) {
    let amount = http_util::query_i64(query, "amount");
    match format_order(amount) {
        None => (400, "{\"error\":\"amount must be non-negative\"}".to_string()),
        Some(body) => (200, body),
    }
}
